Bookmark customTitle from previously bookmarked entry is no longer re-used

bsclifton opened the pull request:
- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Ran `git rebase -i` to squash commits if needed.

Bookmark customTitle from previously bookmarked entry is no longer re-used when re-bookmarking.

Fixes https://github.com/brave/browser-laptop/issues/3284

Auditors: @bbondy @luixxiul, @bradleyrichter

Test Plan:
- Visit a site which is not bookmarked
- Bookmark the site- when the modal comes up, provide your own title "ABC" and hit enter
- Notice bookmark is using customTitle
- Delete bookmark
- Re-add bookmark- this time don't change the title
- Notice bookmark is now properly using title (not the old customTitle)

bbondy commented on js/state/siteUtil.js:
```diff
@@ -109,9 +109,6 @@ module.exports.addSite = function (sites, siteDetail, tag, originalSiteDetail) {
 
-  // We don't want bookmarks and other site info being renamed on users if they already exist
-  // The name should remain the same while it is bookmarked forever.
-  const customTitle = typeof siteDetail.get('customTitle') === 'string' ? siteDetail.get('customTitle') : (siteDetail.get('customTitle') || oldSite && oldSite.get('customTitle'))
```
did you check to make sure this isn't regressed? Use git blame on the task to see the original bug.

bsclifton commented:
Updated code after @bbondy left a comment (see above). The original commit I had done re-introduced an old bug. I reverted that and instead solved the issue by removing the custom title when bookmark is removed. Tests were added to reinforce this behavior :smile:

bbondy commented:
excellent great fix, comment cleanup and tests! 👍
